streams/src/main/java/org/apache/kafka/streams/state/internals/CachingKeyValueStore.java

```diff
@@ -128,9 +134,8 @@ public boolean isOpen() {
     @Override
     public synchronized V get(final K key) {
         validateStoreOpen();
-        if (key == null) {
-            return null;
-        }
+        validateKeyNotNull(key);
```

Comment:
The API for KeyValueStore and ReadOnlyKeyValue store indicates that NPE should be thrown for null keys. This is just applying that consistently.